ui/src/app/kvm/views/KVMList/AddKVM/AddLxd/AddLxd.tsx

```diff
@@ -13,75 +12,69 @@ import AddLxdFields from "./AddLxdFields";
 import FormCard from "app/base/components/FormCard";
 import FormCardButtons from "app/base/components/FormCardButtons";
 import FormikForm from "app/base/components/FormikForm";
-import { useAddMessage } from "app/base/hooks";
-import type { TSFixMe } from "app/base/types";
-import { powerTypes as powerTypesSelectors } from "app/store/general/selectors";
-import { PowerFieldScope } from "app/store/general/types";
-import {
-  formatPowerParameters,
-  generatePowerParametersSchema,
-  useInitialPowerParameters,
-} from "app/store/general/utils";
 import { actions as podActions } from "app/store/pod";
 import podSelectors from "app/store/pod/selectors";
 import { PodType } from "app/store/pod/types";
 import resourcePoolSelectors from "app/store/resourcepool/selectors";
+import type { RootState } from "app/store/root/types";
 import zoneSelectors from "app/store/zone/selectors";
 
 type Props = { setKvmType: SetKvmType };
 
-export type AddLxdValues = { [x: string]: TSFixMe };
+export type AddLxdValues = {
+  name?: string;
+  password?: string;
+  pool: string;
+  power_address: string;
+  project: string;
+  zone: string;
+};
+
+const AddLxdSchema: SchemaOf<AddLxdValues> = Yup.object().shape({
+  name: Yup.string(),
+  password: Yup.string(),
+  pool: Yup.string().required("Resource pool required"),
+  power_address: Yup.string().required("Address is required"),
+  project: Yup.string().required("Project is required"),
+  zone: Yup.string().required("Zone required"),
+});
 
 export const AddLxd = ({ setKvmType }: Props): JSX.Element => {
   const dispatch = useDispatch();
   const history = useHistory();
-  const podSaved = useSelector(podSelectors.saved);
-  const podSaving = useSelector(podSelectors.saving);
-  const podErrors = useSelector(podSelectors.errors);
-  const powerTypes = useSelector(powerTypesSelectors.get);
+  const [lxdAddress, setLxdAddress] = useState("");
+  const projects = useSelector((state: RootState) =>
+    podSelectors.getProjectsByLxdServer(state, lxdAddress)
+  );
+  const errors = useSelector(podSelectors.errors);
+  const saving = useSelector(podSelectors.saving);
   const resourcePools = useSelector(resourcePoolSelectors.all);
   const zones = useSelector(zoneSelectors.all);
   const cleanup = useCallback(() => podActions.cleanup(), []);
-  const [savingPod, setSavingPod] = useState(false);
-  const initialPowerParameters = useInitialPowerParameters();
-
-  useAddMessage(podSaved, cleanup, `${savingPod} added successfully.`, () =>
-    setSavingPod(false)
-  );
-
-  const lxdPowerType = powerTypes.find(
-    (powerType) => powerType.name === PodType.LXD
-  );
-
-  if (!lxdPowerType) {
-    return <Spinner className="u-no-margin u-no-padding" text="Loading" />;
-  }
+  const [authenticating, setAuthenticating] = useState(false);
```

Comment:
Once authenticated this will display another form won't it? I guess this state will end up living at a higher level component which will then display the next form.

Reply:
Actually I thought it'd be doable all in the one component with a few conditionals here and there, but it might be better to establish a pattern that can handle more than one step. Yeah I think I'll end up going with two separate forms - thanks for the sanity check.